### DISCUSSION TITLE
How to calculate fluxes from metabolite concentrations in a medium

### DISCUSSION THREAD

**Discussion by jjoropezav** [micom-dev/micom#115] [Q&A] [answered]
Hi micom developers, 

Thanks for developing an amazing tool. 

Im trying to model a community with micom, i want to create my own medium (which would be made from the metabolite concentrations), i found the discussion where you create a medium/diet using VMH, however if i only have the metabolites, how do i calculate the fluxes? is there any formula or tool that calculate the fluxes from concentrations? 

Heres a row as an example:

Reaction | Metabolite | Concentration
EX_ca2(e) | ca2           |  9.009

Best regards and thanks in advance for the help

**Comment by cdiener** [comment, marked as answer]
Great question and sorry for the late reply. Since I'm lazy and recently replied to a similar question in the cobrapy chat allow me to paste that here:


> Hi Haris, so I'll start by saying that what follows is my take on that and not everybody in the field agrees, so here it goes.
First there is definitely a qualitative relationship since if a metabolite is absent then there can be no uptake flux. Also the highest possible uptake is constrained by the concentration itself since you can't take up more molecules than what are present in the environment.
Flux is concentration per time and per dry weight of the cell. So what is often done is dividing it by the duration of the observation window to get a flux bound. That is what is often discussed since it makes a lot of assumption.
I personally think there is some merit to it, because during exponential growth - which is when the steady state holds - uptake fluxes are a constant and the total uptake during steady state is given as v*t which can't be larger than the abundance of the compound which means v*t <= c, so v <= c/t. The important part to note is that t is the *duration of the steady state* and not the entire observation window and you often don't know that duration. 
Also, that bound is just an upper bound and other factors can influence it so that is not the best upper bound you could find. It also tends to work better in nutrient limited condition.

You can also look at how MICOM media are built by looking at the [MICOM media repository](https://github.com/micom-dev/media). It has notebooks that walk through all steps of the medium creation.
